duviz.py

```diff
@@ -246,6 +246,205 @@ def bar(self, label: str, width: int, fill='-', left='[', right=']', small='|',
         return b
 
 
+class ColumnsRenderer(TreeRenderer):
+    """
+    Render a SizeTree with vertical columns.
+    """
+
+    def __init__(
+        self,
+        height: int,
+        max_depth: int = 5,
+        size_formatter: SizeFormatter = SIZE_FORMATTER_COUNT,
+        color_mode: bool = False,
+    ):
+        super().__init__(max_depth, size_formatter)
+        self.height = height
+        self.color_mode = color_mode
+        if color_mode:
+            self.colorizers = []
+            c = Colorizer()
+            # We use reverse order so that the root is always the same color regardless of the depth
+            for i in reversed(range(max_depth + 1)):
+                cycle = itertools.cycle(
+                    c.COLOR_CYCLE_BMC if i % 2 == 0 else c.COLOR_CYCLE_RGY
+                )
+                self.colorizers.append(cycle)
+
+    def render(self, tree: SizeTree, width: int) -> List[str]:
+        lines = self._render(
+            tree, width // (self.max_depth + 1), self.height, self.max_depth
+        )
+        if lines and not self.color_mode:
+            # The monochrome rendering must be displayed in reverse mode
+            lines[0] = "\x1b[7m" + lines[0]
+            lines[-1] += "\x1b[27m"
+        return lines
+
+    def render_node(
+        self,
+        node: Optional[SizeTree],
+        width: int,
+        height: int,
+        depth: int,
+    ) -> List[str]:
+        """Render a single node"""
+        def _truncate(s, maxlen):
```

Comment:
this can be a top-level helper function I think

Reply:
Done